### PR TITLE
Move async translog sync logic into IndexService

### PR DESCRIPTION
**Pull request by s1monw**
Today the logic to async - commit the translog is in every translog instance
itself. While the setting is a per index setting we manageing it per shard. This
polluts the translog code and can more easily be managed in IndexService.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 1]
```diff
@@ -67,6 +69,8 @@
     private final boolean queryStringAllowLeadingWildcard;
     private final boolean defaultAllowUnmappedFields;
     private final Predicate<String> indexNameMatcher;
+    private volatile Translog.Durabilty durabilty;
```
Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/index/IndexSettings.java]: Maybe we can fix `Durabilty` -> `Durability` (missing `i`)?

File: core/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 2]
```diff
@@ -144,6 +148,10 @@ public IndexSettings(final IndexMetaData indexMetaData, final Settings nodeSetti
         this.parseFieldMatcher = new ParseFieldMatcher(settings);
         this.defaultAllowUnmappedFields = settings.getAsBoolean(ALLOW_UNMAPPED, true);
         this.indexNameMatcher = indexNameMatcher;
+        final String value = settings.get(INDEX_TRANSLOG_DURABILITY, Translog.Durabilty.REQUEST.name());
+        this.durabilty = getFromSettings(settings, Translog.Durabilty.REQUEST);
```
Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/index/IndexSettings.java]: Here too.

File: core/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 3]
```diff
@@ -25,12 +25,11 @@
...
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.index.mapper.internal.AllFieldMapper;
+import org.elasticsearch.index.translog.Translog;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
```
Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/index/IndexSettings.java]: No more wildcard imports right :)

**Comment by mikemccand** [comment]
LGTM, it was unexpected that something as low level as Translog.java would be submitting its own background tasks in a thread pool... it's great to move this higher up.

File: core/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 4]
```diff
@@ -304,4 +322,34 @@ synchronized boolean updateIndexMetaData(IndexMetaData indexMetaData) {
...
+    public Translog.Durability getTranslogDurability() {
+        return durability;
+    }
+
+    private Translog.Durability getFromSettings(Settings settings, Translog.Durability defaultValue) {
+        final String value = settings.get(INDEX_TRANSLOG_DURABILITY, defaultValue.name());
+        try {
+            return Translog.Durability.valueOf(value.toUpperCase(Locale.ROOT));
+        } catch (IllegalArgumentException ex) {
+            logger.warn("Can't apply {} illegal value: {} using {} instead, use one of: {}", INDEX_TRANSLOG_DURABILITY, value, defaultValue, Arrays.toString(Translog.Durability.values()));
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/IndexSettings.java]: should we log the exception to have the stack trace?

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/IndexSettings.java]: oh nevermind I thought it was a parsing method, but this is the simple Enum.valueOf

**Comment by jpountz** [comment]
+1

**Pull request merged**